presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -157,26 +159,29 @@ else if (type.equals(BIGINT) && (statistics instanceof LongStatistics || statist
             }
             return createDomain(type, hasNullValue, parquetIntegerStatistics);
         }
-        else if (type.equals(DOUBLE) && (statistics instanceof DoubleStatistics || statistics instanceof FloatStatistics)) {
-            ParquetDoubleStatistics parquetDoubleStatistics;
-            if (statistics instanceof DoubleStatistics) {
-                DoubleStatistics doubleStatistics = (DoubleStatistics) statistics;
-                // ignore corrupted statistics
-                if (doubleStatistics.genericGetMin() > doubleStatistics.genericGetMax()) {
-                    return Domain.create(ValueSet.all(type), hasNullValue);
-                }
-                parquetDoubleStatistics = new ParquetDoubleStatistics(doubleStatistics.genericGetMin(), doubleStatistics.genericGetMax());
-            }
-            else {
-                FloatStatistics floatStatistics = (FloatStatistics) statistics;
-                // ignore corrupted statistics
-                if (floatStatistics.genericGetMin() > floatStatistics.genericGetMax()) {
-                    return Domain.create(ValueSet.all(type), hasNullValue);
-                }
-                parquetDoubleStatistics = new ParquetDoubleStatistics((double) floatStatistics.getMin(), (double) floatStatistics.getMax());
+        else if (type.equals(DOUBLE) && statistics instanceof DoubleStatistics) {
+            DoubleStatistics doubleStatistics = (DoubleStatistics) statistics;
+            // ignore corrupted statistics
+            if (doubleStatistics.genericGetMin() > doubleStatistics.genericGetMax()) {
+                return Domain.create(ValueSet.all(type), hasNullValue);
             }
+            ParquetDoubleStatistics parquetDoubleStatistics = new ParquetDoubleStatistics(doubleStatistics.genericGetMin(), doubleStatistics.genericGetMax());
             return createDomain(type, hasNullValue, parquetDoubleStatistics);
         }
+        else if (type.equals(FLOAT) && statistics instanceof FloatStatistics) {
+            FloatStatistics floatStatistics = (FloatStatistics) statistics;
+            // ignore corrupted statistics
+            if (floatStatistics.genericGetMin() > floatStatistics.genericGetMax()) {
+                return Domain.create(ValueSet.all(type), hasNullValue);
+            }
+
+            ParquetIntegerStatistics parquetStatistics = new ParquetIntegerStatistics(
```

Comment:
I've just confirmed that the TupleDomain contains the float-to-raw-int-bits Long value. This seems not very user friendly, but I don't know how we can change that without special casing everything for float.